### PR TITLE
Task/Update log format on dev and prod

### PR DESCRIPTION
**Pull request by pcraig3**
# 🌲 LOGGING!! 🌲 

We love it!

Some pretty big changes here, or maybe small changes, depending on your perspective. 

Long & short:
- All logs are `stdout` or `stderr`
- All logs are JSON
- No logs in cypress
- Removed `winston` logger
- You can delete your `/logs` directory
- Stopped logging all CSS, JS, img files, etc.

//

Basically, after thinking about it for a while, I really had only a couple of constraints. 

1. All logs should go to `stdout`
2. All logs should be JSON

## Everything `stdout`

[According to Docker](https://docs.docker.com/config/containers/logging/), "By default, docker logs or docker service logs shows the command’s output just as it would appear if you ran the command interactively in a terminal." So, we don't want to be creating big log files using winston, which is what we were doing. So I got rid of logging to files, which means you can delete your `/logs` directory (should be in the project root).

## Everything JSON

The decision to do JSON is based on past experience trying to search through logs for things. JSON logs are generally easier because you can find keys and add new ones whenever you want without having to do weird string parsing because of some unexpected formatting change. 

winston can handle JSON, yes, but morgan seemed simpler and faster to work with. So basically, to get morgan to use JSON (which it doesn't ordinarily do) I used this gist: https://gist.github.com/cgmartin/913bb12097ff07132597 

Once I got morgan the way I wanted it, I removed `winston`. We can add it back later if we need something it does but let's not overcomplicate things at this point. 

## What about the format?

No strong opinions. We want to have useful prod logs but not overwhelming dev logs. Check the [`morgan.config.js`](https://github.com/cds-snc/cra-claim-tax-benefits/compare/task/log-a-log?expand=1#diff-459318efbac66a7d933c8146432e3785) file for formatting. Happy to change it if needed.

## What about errors?

Errors were a bit weird. I wanted to keep all the error stuff in there, but push them into our JSON logs because I don't want them to get lost if we end up having to debug weird prod issues. 

So the error behaviour is the default express thing: 
- log them to the terminal

On dev:
- also put the stack trace on the screen

On prod:
- don't put the stack trace on the screen
- put the stack trace in our JSON logs

# Dev logs before/after

Before: dev logs for the first 2 pages (1 bad request).

```
2019-10-15T17:46:56.306Z [cra-ctb] info: GET / 302 17.064 ms - 56

2019-10-15T17:46:56.656Z [cra-ctb] info: GET /start 304 335.064 ms - -

2019-10-15T17:46:56.726Z [cra-ctb] info: GET /img/sig-blk-en.svg 304 6.022 ms - -

2019-10-15T17:46:56.748Z [cra-ctb] info: GET /img/wmms-blk.svg 304 20.889 ms - -

2019-10-15T17:46:56.753Z [cra-ctb] info: GET /styles.css 200 47.457 ms - -

2019-10-15T17:46:56.927Z [cra-ctb] info: GET /js/script.js 304 1.077 ms - -

2019-10-15T17:46:59.562Z [cra-ctb] info: GET /login/code 200 118.011 ms - -

2019-10-15T17:46:59.639Z [cra-ctb] info: GET /img/wmms-blk.svg 304 13.869 ms - -

2019-10-15T17:46:59.642Z [cra-ctb] info: GET /img/sig-blk-en.svg 304 10.805 ms - -

2019-10-15T17:46:59.663Z [cra-ctb] info: GET /styles.css 200 55.178 ms - 21441

2019-10-15T17:46:59.709Z [cra-ctb] info: GET /js/script.js 304 0.544 ms - -

2019-10-15T17:47:02.778Z [cra-ctb] info: POST /login/code 422 147.535 ms - -

2019-10-15T17:47:02.840Z [cra-ctb] info: GET /img/sig-blk-en.svg 304 1.394 ms - -

2019-10-15T17:47:02.841Z [cra-ctb] info: GET /img/wmms-blk.svg 304 1.423 ms - -

2019-10-15T17:47:02.853Z [cra-ctb] info: GET /styles.css 304 15.280 ms - -

2019-10-15T17:47:02.912Z [cra-ctb] info: GET /js/script.js 304 4.132 ms - -

2019-10-15T17:47:04.610Z [cra-ctb] info: POST /login/code 302 6.317 ms - 64

2019-10-15T17:47:04.712Z [cra-ctb] info: GET /login/sin 304 92.316 ms - -

2019-10-15T17:47:04.774Z [cra-ctb] info: GET /img/sig-blk-en.svg 304 1.351 ms - -

2019-10-15T17:47:04.774Z [cra-ctb] info: GET /img/wmms-blk.svg 304 1.488 ms - -

2019-10-15T17:47:04.778Z [cra-ctb] info: GET /styles.css 304 5.650 ms - -

2019-10-15T17:47:04.816Z [cra-ctb] info: GET /js/script.js 304 0.504 ms - -
```

After: dev logs for the first 2 pages (1 bad request).

```
{"method":"GET","url":"/","status":"302","response-time":"11.704ms"}
{"method":"GET","url":"/start","status":"304","response-time":"339.817ms"}
{"method":"GET","url":"/login/code","status":"200","response-time":"117.774ms"}
{"method":"POST","url":"/login/code","status":"422","response-time":"98.978ms"}
{"method":"POST","url":"/login/code","status":"302","response-time":"2.509ms"}
{"method":"GET","url":"/login/sin","status":"304","response-time":"77.601ms"}
``` 

# Prod logs before/after

Before: prod logs for the first 2 pages (1 bad request).

```
(no logs at all to std out)
```

After: prod logs for the first 2 pages (1 bad request).

```
{"method":"GET","url":"/","status":"302","response-time":"8.816ms","timestamp":"2019-10-15T17:50:44.180Z","content-length":"56","user-agent":"Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_0) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/77.0.3865.120 Safari/537.36"}
{"method":"GET","url":"/start","status":"304","response-time":"308.849ms","timestamp":"2019-10-15T17:50:44.496Z","user-agent":"Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_0) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/77.0.3865.120 Safari/537.36"}
{"method":"GET","url":"/login/code","status":"200","response-time":"86.101ms","timestamp":"2019-10-15T17:50:46.681Z","referrer":"http://localhost:3005/start","user-agent":"Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_0) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/77.0.3865.120 Safari/537.36"}
{"method":"POST","url":"/login/code","status":"422","response-time":"8.940ms","timestamp":"2019-10-15T17:50:49.017Z","referrer":"http://localhost:3005/login/code","user-agent":"Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_0) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/77.0.3865.120 Safari/537.36"}
{"method":"POST","url":"/login/code","status":"302","response-time":"1.112ms","timestamp":"2019-10-15T17:50:51.164Z","content-length":"64","referrer":"http://localhost:3005/login/code","user-agent":"Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_0) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/77.0.3865.120 Safari/537.36"}
{"method":"GET","url":"/login/sin","status":"304","response-time":"80.404ms","timestamp":"2019-10-15T17:50:51.256Z","referrer":"http://localhost:3005/login/code","user-agent":"Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_0) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/77.0.3865.120 Safari/537.36"}
```

### CODE REVIEW

File: app.js [hunk 1]
```diff
@@ -113,17 +110,23 @@ app.use(function(req, res, next) {
...
-// handle global errors.
-app.use(function(err, req, res) {
-  res.locals.message = err.message
-  res.locals.error = req.app.get('env') === 'development' ? err : {}
+// Pass error information to res.locals
+app.use((err, req, res, next) => {
+  let errObj = {}
+
+  let status = err.status || err.statusCode || 500
+  status = status < 400 ? 500 : status
```
Comment (katedee) [inline app.js]: 🤔 doesn't this mean we'll be filtering out certain 400 errors, like 431, which we got before about our header size

Comment (katedee) [inline app.js]: Wouldn't this set some of our 200, 300 status codes to 500? 🤔

Comment (pcraig3) [reply app.js]: Okay! Fair enough: https://github.com/cds-snc/cra-claim-tax-benefits/pull/254/commits/75c1b23ef865942ed6c4c3b2bf4a09a0cb850320

Comment (charlesmorin) [reply app.js]: I asked myself the same question but now I understand the behavior. Looks good to me then.

**Comment by charlesmorin** [comment]
Looks good to me. Merge away!

**Pull request merged**